### ISSUE TITLE
[feature request] export .ics-file upon change for the purpose of backup and sync

### ISSUE THREAD

**Issue by shanks1337** [SimpleMobileTools/Simple-Calendar#1209]
For my password manager I sync the password file via "syncthing" between my devices so all of them are up to date and the file won't be lost if one device breaks. It would be great if Simple Calendar Pro had similar functionality. I see that the use case between a calendar and password manager is different but at least for backup purposes I'd say it makes sense. At the moment the ics-file is lost when my phone breaks and I usually don't have a recent export of the ics-file. It would not make Simple Calendar Pro any less simple because most users would not notice the difference.

I don't use any online backup besides my own computers. So for my use case I need to have a file which can be synced between devices. Synchronisations via CalDAV are no proper backup since has no versioning and could be theoretically all be over written by malware. With proper backup this is not the case.

If you consider this then I suggest the behavior as following:
- export .ics-file upon change in Simple Calendar Pro. _Not_ periodically since it could break another calendar programm, if it was changed by during the period. In a single user setup it's far more unlikely to break it upon change. 
- the most recent ics-file would be always "calendar.ics" in "/storage/emulated/0/Simple Calendar Pro/" so it can be accessed by other apps.
- if a change was made in Simple Calendar Pro then rename old "calendar.ics" to "YYYY-MM-DD-hh-mm-ss-calendar.ics" and write the most recent as "calendar.ics". Very similar to how WhatsApp is doing it.

I would love to see Simple Calendar Pro have this feature. Next step would be Simple Calendar detecting change in the .ics-file and reload it.

**Comment by FOSS-Love** [comment]
Yes this is most awaiting feature for me, i also felt the need of this feature but not only for calendar, this needs to be implemented for all supported backups of simple mobile tools, Though i dont stress on having a file versioning as Syncthing has a built in versioning system. But Automatic backup of everthing will be great.

I dont know the technical possibility but this is most awaiting feature for simple mobile tools.

**Comment by FOSS-Love** [comment]
I have checked for the implementation of this feature, and i found an application that does something similar.

Check out series giude app
https://github.com/UweTrottmann/SeriesGuide.

the above application implements the automatic backup.

Dont know how simliarity it has, but that is how the feature has to work i guess.

**Comment by adapt-L** [comment]
I'd like to offer a bounty for anyone who can implement this feature, 75 dollars in the implementer's cryptocurrency of choice (XMR, BTC, BCH, LTC). If you don't trust me to payout, I am willing to fund a 2-of-2 multisig with the main developer until the feature is implemented.

My requirements are that the user is able to choose some option like "Sync to .ics file". This would prompt the user to create a new file that the app will write all calendar changes to. It would also run in the background and/or listen for changes to the .ics file and update the user's calendar appropriately.

This feature would be great because it would let me sync my calendar between my computers without a third party service. It seems that GNOME calendar already supports this behavior. However I'm not interested in patching this myself.

**Comment by ReaderGuy42** [comment]
Any updates on this?

**Comment by adapt-L** [comment]
Holy cow, I can't believe I didn't figure this out earlier. All you have to do is use this app called ICSx5: https://icsx5.bitfire.at/ (It's open source and on fdroid). It is basically the same thing as DAVx5 but for .ics calendars instead of CalDAV. I found it while reading this: https://www.davx5.com/faq/subscribe-ics-file

I can write up a guide for you guys, but it is really easy to use with syncthing. ICSx5 will sync the calendar to the .ics file, and syncthing can sync it with your laptop. The only snag is that you have to use a calendar on the laptop side that can read and write to local .ics files. For example, gnome calendar (https://apps.gnome.org/Calendar/) works but mozilla thunderbird doesn't.

**Comment by adapt-L** [comment]
Forget about the bug bounty, I'll just make a donation to ICSx5

**Comment by shanks1337** [comment]
It seems like this feature or at least the back up part was somewhat implemented in an update from may(?). There is now an option to have _periodic_ backups (hardcoded daily to 6:00 AM) instead of upon change.

see: https://github.com/SimpleMobileTools/Simple-Calendar/pull/2027

**Comment by tibbi** [comment]
ye, but it is a periodic export, not triggered by event changes
